### PR TITLE
Enable Mergify support

### PR DESCRIPTION
**Pull request by Gabriel439**
Related to https://github.com/dhall-lang/dhall-haskell/issues/1058

This is an attempt to test drive the Mergify GitHub app.  Unfortunately, the
only way to know for sure if the app works is to merge this configuration
into `master`, but if it does not work out then I will follow up with a pull
request to revert this change.

For more details about how Mergify works, see: https://doc.mergify.io/getting-started.html#configuration

### CODE REVIEW

**Comment by sjakobi** [comment]
So even an approved PR will still require a `merge me` label so `Mergify` can pick it up? I like that, as sometimes you want to keep an approved PR open for a bit to allow more reviews or discussion.

LGTM then.

File: .mergify.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+- actions:
+    merge:
+      method: squash
+  name: Automatically merge pull requests
+  conditions:
+  - status-success=continuous-integration/appveyor/pr
+  - status-success=hydra
+  - label=merge me
+  - ! '#approved-reviews-by>=1'
+  - ! '#changes-requested-reviews-by=0'
```
Comment (ocharles) [inline .mergify.yml]: The irony of having `.yml` checked in :laughing:

File: .mergify.yml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+pull_request_rules:
+- actions:
+    merge:
+      method: squash
```
Comment (ocharles) [inline .mergify.yml]: Do you also need to add `strict: true`? (https://doc.mergify.io/strict-workflow.html)

File: .mergify.yml (hunk 1 above)
Comment (Gabriel439) [reply .mergify.yml]: To be fair, I generated it with `dhall-to-yaml` 🙂

**Pull request merged**